sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -43,6 +61,45 @@ export abstract class SwapRouter {
     })
   }
 
+  /**
+   * Builds the call parameters for a migration from a V3 position to a V4 position.
+   * Some requirements of the parameters:
+   *   - v3RemoveLiquidityOptions.collectOptions.recipient must equal v4PositionManager
+   *   - v3RemoveLiquidityOptions.liquidityPercentage must be 100%
+   *   - input pool and output pool must have the same tokens
+   *   - V3 NFT must be approved, or valid inputV3NFTPermit must be provided with UR as spender
+   */
+  public static migrateV3ToV4CallParameters(options: MigrateV3ToV4Options): MethodParameters {
+    const token0 = options.inputPosition.pool.token0
+    const token1 = options.inputPosition.pool.token1
+    invariant(token0 === options.outputPosition.pool.token0, 'TOKEN0_MISMATCH')
+    invariant(token1 === options.outputPosition.pool.token1, 'TOKEN1_MISMATCH')
+    invariant(options.v3RemoveLiquidityOptions.liquidityPercentage.equalTo(new Percent(100)), 'FULL_REMOVAL_REQUIRED')
+
+    const planner = new RoutePlanner()
+
+    if (options.inputV3NFTPermit) {
+      // note: permit spender should be UR
```

Comment:
yeah we should be checking the spender here in the invariants above

Reply:
how can i get the UR address to test?